File: mac/microcosmMac2/Models/PlanetData.cs
using System;
using System.Collections.Generic;

namespace microcosmMac2.Models
{
    public class PlanetData
    {
        // swiss_epheと対応した惑星番号
        public int no { get; set; }

        // 0.0～359.99、絶対値
        public double absolute_position { get; set; }

        // 速度(マイナスなら逆行)
        public double speed { get; set; }

        // アスペクト対象リスト
        public List<AspectInfo> aspects;
        // Pとのアスペクトはここ
        public List<AspectInfo> secondAspects;
        // Tとのアスペクトはここ
        public List<AspectInfo> thirdAspects;

        // 感受点はtrue
        public bool sensitive;

        // 天体を表示するか
        public bool isDisp = true;
        // アスペクトを表示するか
        public bool isAspectDisp = true;

        // PlanetData[1]のaspects1 = P-P
        //public List<AspectInfo> aspects0;
        //public List<AspectInfo> aspects1;
        //public List<AspectInfo> aspects2;
        //public List<AspectInfo> aspects3;
        //public List<AspectInfo> aspects4;
        //public List<AspectInfo> aspects5;
        //public List<AspectInfo> aspects6;

        public PlanetData()
        {
        }
    }
}

File: microcosm-swift-git/microcosm-swift-git/Model/PlanetData.swift
//
//  PlanetData.swift
//  microcosm-swift-1
//
//  Created by 緒形雄二 on 2024/07/17.
//

import Foundation

public class PlanetData
{
    // swiss_epheと対応した惑星番号
    public var no: Int
    // 0.0～359.99、絶対値
    public var absolute_position: Double
    // 速度(マイナスなら逆行)
    public var speed: Double;
    // 感受点はtrue
    public var sensitive: Bool
    // アスペクト対象リスト
    public var aspects: [AspectInfo] = [AspectInfo]()
    // Pとのアスペクトはここ
    public var secondAspects: [AspectInfo] = [AspectInfo]()
    // Tとのアスペクトはここ
    public var thirdAspects: [AspectInfo] = [AspectInfo]()

    // 天体を表示するか
    public var isDisp: Bool = true
    // アスペクトを表示するか
    public var isAspectDisp: Bool = true

    init()
    {
        no = -1
        absolute_position = 0
        speed = 0
        sensitive = false
    }
}
